src/api/server.go

```diff
@@ -30,13 +24,15 @@ type Server struct {
 	db           *DbRunner
 	thrift_proto string
 	thrift_addr  string
+	bind_info    *bindInfo
 }
 
-func ListenAndServe(addr, thrift_addr, thrift_proto, dbdriver, dbconn string) (*Server, error) {
+func ListenAndServe(addr, thrift_addr, thrift_proto, dbdriver, dbconn, v4, v6 string) (*Server, error) {
```

Comment:
would that work if v6 is null or v4 is null? We should account for cases that are v4 or v6 only.